game/src/contrib/systems/CollisionSystem.java

```diff
@@ -43,6 +74,11 @@ private CollisionData buildData(Entity a, Entity b) {
         return new CollisionData(cca, ccb);
     }
 
+    /**
+     * Checks wheter a new Collision is happening or wheter a Collision ended
```

Comment:
```suggestion
     * Checks whether a new collision is happening or whether a collision is ended
```

Reply:
die funktion prüft doch bei onenter und onleave (was genau entered oder leaved da was?), ob eine kollision vorliegt? das kommt aus der beschreibung nicht raus.

Reply:
Hier habe ich ebenfalls versucht die Doku einmal verständlicher zu schreiben und auch Kommentare im Code zur besseren Verständlichkeit.

Reply:
hmmm, dazu muss man jetzt vermutlich die kollisionsprüfung komplett auf dem schirm haben. naiv hätte ich immer zwei entitäten verglichen, also ihre hitboxen, und bei überlappung der hitboxen hat man eine kollision. 

die ansicht im browser ist ein wenig unübersichtlich, aber es hat den anschein, dass du dir kollisionen intern im collisionsystem merkst - und das sind die begrifflichkeiten für "is happening" und "is ended"?

Reply:
Also `checkForCollision(hitbox1, hitbox2) : boolean` ist die Kontrolle ob derzeit eine Kollision vorhanden ist. Also ob irgendeine der Ecken in der Fläche der anderen Hitbox ist. Bei unserem Check wird vorausgesetzt, dass die Hitboxen nicht gedreht um irgendeine Achse sind. 

Wenn wir also nun eine Hitbox mit einer anderen vergleichen und ein True zurückkommt, ist aktuell eine Kollision vorhanden (collision is happening). Da wir irgendwann mal uns dafür entschieden hatten, nur einmal die Hitbox über eine Kollision zu informieren, gibt es eine Map von allen bisher aktiven Kollisionen. Wenn die Kollision noch nicht in dieser Map vorhanden ist, wird ein Eintrag in der Map erstellt und die beiden Hitboxen informiert. Dabei hatten wir `onEnter(other : Collidecomponent, Direction)` bei Beginn einer Kollision geeinigt. Vielleicht wäre ein `onBegin` oder `onCollisionBegin` besser. 

Der zweite Teil ist die Kontrolle, ob die Kollision zuvor vorhanden war und nun zu Ende ist. In diesem Fall wird aus der Map der aktuellen Kollisionen der Eintrag gelöscht und die `onLeave(other: CollideComponent, Direction dir)` aufgerufen. Hier wird die Kollision verlassen oder beendet.

Reply:
also war meine interpretation vom kommentar drüber nicht ganz falsch :). 

die frage ist, ob wir es jetzt hier bei der version des kommentars belassen oder noch einen teil der erklärung mit aufnehmen (ich wäre für letzteres). zusätzlich könnte die gesamte erklärung noch in das klassen-javadoc rein.